docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -1630,37 +1626,57 @@ DARs. They should both succeed again:
 
   > cd v1/my-pkg
   > daml build
-  > daml sandbox upload-dar --port 6865
+  > daml ledger upload-dar --port 6865
   ...
   Uploading .daml/dist/my-pkg-1.0.0.dar to localhost:6865
   DAR upload succeeded.
   > cd ../../v2/my-pkg
   > daml build
-  > daml sandbox upload-dar --port 6865
+  > daml ledger upload-dar --port 6865
   ...
   Uploading .daml/dist/my-pkg-1.1.0.dar to localhost:6865
   DAR upload succeeded.
 
 Packages with new versions cannot remove an instance that is already
 there. For example, the v2 IOU template cannot remove its instance of
-``HasValue``. Remove the interface instance for ``HasValue`` from
+``HasValue``. Comment out the interface instance for ``HasValue`` from
 ``v2/my-pkg/daml/Main.daml`` completely, then restart the sandbox and try to
 reupload the two versions:
 
 .. code:: bash
 
   > cd v1/my-pkg
   > daml build
-  > daml sandbox upload-dar --port 6865
+  > daml ledger upload-dar --port 6865
   ...
   Uploading .daml/dist/my-pkg-1.0.0.dar to localhost:6865
   DAR upload succeeded.
   > cd ../../v2/my-pkg
   > daml build
-  > daml sandbox upload-dar --port 6865
+  > daml ledger upload-dar --port 6865
+  ...
+  Uploading .daml/dist/my-pkg-2.0.0.dar to localhost:6865
+  upload-dar did not succeed: ... Implementation of interface ...:MyIface:HasValue by template IOU appears in package that is being upgraded, but does not appear in this package.
+
+Packages with new versions cannot add an interface instance to an existing
```

Comment:
Hi @paulbrauner-da , under "Best Practices" > "Remove Retroactive Instances", it still states that "retroactive interface instances should be moved to newer versions of templates, such that changes to the instance warrants a new version of the template".

Doesn't it conflict with the new restriction here that "packages with new versions cannot add an interface instance to an existing template"?

Reply:
Thank you for pointing this out, we'll fix the inconsistency.